bridge/server/utils/configuration.ts

```diff
@@ -1,37 +1,226 @@
+  enabled: boolean;
+  discoveryURL: string;
+  baseURL: string;
+  clientID: string;
+  clientSecret?: string;
+  scope: string;
+  tokenAlgorithm: string;
+  allowedLogoutURL: string;
+  nameProperty?: string;
+  session: OAuthSessionConfig;
+}
+
+export interface OAuthSessionConfig {
+  secureCookie: boolean;
+  trustProxyHops: number;
+  timeoutMin: number;
+  validationTimeoutMin: number;
+}
+
+export interface APIConfig {
+  url: string;
+  token: string | undefined;
+  showToken: boolean;
+}
+
+export interface FeatureConfig {
+  pageSize: PageSizeConfiguration;
+  installationType: string;
+  automaticProvisioningMessage: string;
+  prefixPath: string;
+  configDir: string;
+  versionCheck: boolean;
+}
+
+export interface PageSizeConfiguration {
+  project: number;
+  service: number;
+}
+
+export interface URLsConfig {
+  lookAndFeel?: string;
+  integrationPage: string;
+  CLI: string;
+}
+
+export interface MongoConfig {
+  user: string;
+  password: string;
+  host: string;
+  db: string;
+}
+
+/**
+ * Env var list
+ */
+
+export enum EnvVar {
+  LOGGING_COMPONENTS = 'LOGGING_COMPONENTS',
+  SHOW_API_TOKEN = 'SHOW_API_TOKEN',
+  API_URL = 'API_URL',
+  API_TOKEN = 'API_TOKEN',
+  AUTH_MSG = 'AUTH_MSG',
+  BASIC_AUTH_USERNAME = 'BASIC_AUTH_USERNAME',
+  BASIC_AUTH_PASSWORD = 'BASIC_AUTH_PASSWORD',
+  REQUEST_TIME_LIMIT = 'REQUEST_TIME_LIMIT',
+  REQUESTS_WITHIN_TIME = 'REQUESTS_WITHIN_TIME',
+  CLEAN_BUCKET_INTERVAL = 'CLEAN_BUCKET_INTERVAL',
+  OAUTH_ALLOWED_LOGOUT_URLS = 'OAUTH_ALLOWED_LOGOUT_URLS',
+  OAUTH_BASE_URL = 'OAUTH_BASE_URL',
+  OAUTH_CLIENT_ID = 'OAUTH_CLIENT_ID',
+  OAUTH_CLIENT_SECRET = 'OAUTH_CLIENT_SECRET',
+  OAUTH_DISCOVERY = 'OAUTH_DISCOVERY',
+  OAUTH_ENABLED = 'OAUTH_ENABLED',
+  OAUTH_NAME_PROPERTY = 'OAUTH_NAME_PROPERTY',
+  OAUTH_SCOPE = 'OAUTH_SCOPE',
+  SECURE_COOKIE = 'SECURE_COOKIE',
+  SESSION_TIMEOUT_MIN = 'SESSION_TIMEOUT_MIN',
+  TRUST_PROXY = 'TRUST_PROXY',
+  SESSION_VALIDATING_TIMEOUT_MIN = 'SESSION_VALIDATING_TIMEOUT_MIN',
+  OAUTH_ID_TOKEN_ALG = 'OAUTH_ID_TOKEN_ALG',
+  CLI_DOWNLOAD_LINK = 'CLI_DOWNLOAD_LINK',
+  INTEGRATIONS_PAGE_LINK = 'INTEGRATIONS_PAGE_LINK',
+  LOOK_AND_FEEL_URL = 'LOOK_AND_FEEL_URL',
+  AUTOMATIC_PROVISIONING_MSG = 'AUTOMATIC_PROVISIONING_MSG',
+  CONFIG_DIR = 'CONFIG_DIR',
+  KEPTN_INSTALLATION_TYPE = 'KEPTN_INSTALLATION_TYPE',
+  PROJECTS_PAGE_SIZE = 'PROJECTS_PAGE_SIZE',
+  SERVICES_PAGE_SIZE = 'SERVICES_PAGE_SIZE',
+  PREFIX_PATH = 'PREFIX_PATH',
+  ENABLE_VERSION_CHECK = 'ENABLE_VERSION_CHECK',
+  MONGODB_DATABASE = 'MONGODB_DATABASE',
+  MONGODB_HOST = 'MONGODB_HOST',
+  MONGODB_PASSWORD = 'MONGODB_PASSWORD',
+  MONGODB_USER = 'MONGODB_USER',
+  NODE_ENV = 'NODE_ENV',
+  VERSION = 'VERSION',
+}
+
+const _componentName = 'Configuration';
+
 /**
  * @param options Customization options that override env var options.
  * @returns Returns the Bridge-server configuration.
  */
 export function getConfiguration(options?: BridgeOption): BridgeConfiguration {
```

Comment:
You are right, options are for test purposes where we don't want to mess around with env var but instead use a declarative approach that can lead us to run tests in parallel 😉

Reply:
So does anything speak against this approach?
```ts
getConfiguration(config: Record<EnvVar, string | undefined>): BridgeConfiguration {...}
```
This would remove the `BridgeOption` and we would have one object instead of two for fetching/setting the config.

Reply:
Maybe I was not clear enough: your suggestion is valid :) I will address it tomorrow if I manage to reserve some time 😭

Reply:
That's a good suggestion, Klaus, this would make the usage easier for the tests. What I like with the other approach though is that we have the same structure for the defaults as for the actual configuration. No real preference, to be honest, so I'd say keep it as it is.

Reply:
We could keep the logical grouping of conf parameters if we introduce a method that transforms `process.env` to `BridgeOption`